ERCS/erc-7586.md

```diff
@@ -0,0 +1,195 @@
+
+    /**
+    *  @notice Returns the floating rate spread, i.e. the fixed part of the floating interest rate. It is RECOMMENDED to express the spread in basis point unit
+    *          1 basis point = 0.01% = 0.0001
+    *          ex: if spread = 0.5%, then floatingRateSpread() => 50 basis points
+    *
+    *          floatingRate = benchmark + spread
+    */
+    function spread() external view returns(uint256);
+
+    /**
+    * @notice Returns the contract address of the asset to be transferred when swapping IRS. Depending on what the two parties aggreed upon, this could be a currency, etc.
+    *         Example: If the two parties agreed to swap interest rates in USDC, then this function should return the USDC contract address.
+    *                  This address SHOULD be used in the `swap` function to transfer the interest difference to either the `payer` or the `receiver`. Example: IERC(assetContract).transfer
+    */
+    function assetContract() external view returns(address);
+
+    /**
+    *  @notice Returns the notional amount in unit of asset to be transferred when swapping IRS. This amount serves as the basis for calculating the interest payments, and may not be exchanged
+    *          Example: If the two parties aggreed to swap interest rates in USDC, then the notional amount may be equal to 1,000,000 USDC, etc 
+    */
+    function notionalAmount() external view returns(uint256);
+
+    /**
+    *  @notice Returns the interest payment frequency
+    */
+    function paymentFrequency() external view returns(uint256);
+
+    /**
+    *  @notice Returns an array of specific dates on which the interest payments are exchanged. Each date MUST be a Unix timestamp like the one returned by block.timestamp
+    *          The length of the array returned by this function MUST equal the total number of swaps that should be realized
+    *
+    *  OPTIONAL
+    */
+    function paymentDates() external view returns(uint256[] memory);
+
+    /**
+    *  @notice Returns the starting date of the swap contract. This is a Unix Timestamp like the one returned by block.timestamp
+    */
+    function startingDate() external view returns(uint256);
+
+    /**
+    *  @notice Returns the maturity date of the swap contract. This is a Unix Timestamp like the one returned by block.timestamp
+    */
+    function maturityDate() external view returns(uint256);
+
+    /**
+    *  @notice Returns the benchmark in basis point unit
+    *          Example: value of one the following rates: CF BIRC, EURIBOR, HIBOR, SHIBOR, SOFR, SONIA, TONAR, etc.
+    */
+    function benchmark() external view returns(uint256);
+
+    /**
+    *  @notice Returns the oracle contract address for the benchmark rate.
+    *          This contract SHOULD be used to fetch real time benchmark rate
+    *          Example: Contract address for `CF BIRC`
+    *
+    *  OPTIONAL. The two parties MAY agree to set the benchmark manually
```

Comment:
Should specify what this function returns when two parties set the benchmark manually (I'd assume the zero address, or revert maybe?)

Reply:
Sure! It should return the zero address in case the benchmark is set manually